### ISSUE TITLE
cachedPatch should pick up previous patch.

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#2327]
1. If no patch exist in cwd then download + use
2. If exists, use and then download

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "minor"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.25.0](https://github.com/open-wa/wa-automate-nodejs/compare/4.24.2...4.25.0) (2021-12-02)

- 🥚 patches update fixes #2321 [`#2321`](https://github.com/open-wa/wa-automate-nodejs/issues/2321)
- Release 4.25.0 [`0a26fb5`](https://github.com/open-wa/wa-automate-nodejs/commit/0a26fb5afae1f01cbbe0fd8a6d68bef3eb0f1ed4)
- ✨ feat + refactor: refactored patch code to `patch_manager` & improved `config.cachedPatch` #2327 [`45c5462`](https://github.com/open-wa/wa-automate-nodejs/commit/45c546245aea454f86bb3094519fa5ba0535e955)
- ♻️ refactor: `process.send` util #2324 [`870303c`](https://github.com/open-wa/wa-automate-nodejs/commit/870303cc01db42b0e8c04af569db6ef3a09b0578)
- 🚧 WIP: prelim `Dialog` wizard work #1794 [`8ecad06`](https://github.com/open-wa/wa-automate-nodejs/commit/8ecad06c9ca8276c08055cf6da403d707aff2454)
- ⚡️ perf: save 2 seconds on first qr code #2326 [`c9528b1`](https://github.com/open-wa/wa-automate-nodejs/commit/c9528b1da137963519e4cf7e95bb9ec9b723ba45)
- ✨ feat: `config.ghPatch` - use GH link for patches #2327 [`2a2453e`](https://github.com/open-wa/wa-automate-nodejs/commit/2a2453e838e1f7ca0a9184a864d06c83bde3f691)
- 🔧 launch: send `QR` via process on new qr code #2324 [`c7fd3f1`](https://github.com/open-wa/wa-automate-nodejs/commit/c7fd3f10c5f6b74b86067bca3ab2bb728cba2ddb)
- 🥚  patches update [`66e01fe`](https://github.com/open-wa/wa-automate-nodejs/commit/66e01fe7b107ac3a84c5ce80da2ba60d60309fde)
- 🥚  patches update [`b9cd38c`](https://github.com/open-wa/wa-automate-nodejs/commit/b9cd38c27d34da7b5476b6f376d6ecd0a8905090)
- 🥚 patches update [`5115eb8`](https://github.com/open-wa/wa-automate-nodejs/commit/5115eb8613720b189a7d20564c50f8ed393fa3b8)
- updated types-only package [`adfbd08`](https://github.com/open-wa/wa-automate-nodejs/commit/adfbd08d76c96e45373f2b59593b9d201f2626cd)
